### PR TITLE
Restructures package to be go get'able with standard Go command/package structures

### PR DESCRIPTION
**Pull request by prologic**
This does a few things:

- Makes this package `go get`-able which is nice because some of us like ot just `go get` stuff -- Even if its just a binary/tool/app :)
- Moves the "internal" code into the standard Go `internal` package structure so its nicely hidden away in any go doc references (_I doubt you want to export anything at this time?_)
- Rejigs the `Makefile` accordingly (_considered using `goreleaser` to manage your releases?_)

Thanks!

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -2,24 +2,24 @@ all: build release clean
 
 VERSION=`git tag |tail -n1`
 build:
-	@cd src && go build -ldflags "-s -w -X main.version=${VERSION}" .
+	@go build -ldflags "-s -w -X $(shell go list)/internal.Version=${VERSION}" ./cmd/gorss/...
```
Comment (Lallassu) [inline Makefile]: cmd doesn't exists here. Not sure what the intent is? Why build to an `cmd/gorss` directory?

File: src/main.go [hunk 2]
```diff
@@ -1,83 +0,0 @@
-package main
```
Comment (Lallassu) [inline src/main.go]: Main removed, kinda hard to run the application then? :)

**Comment by prologic** [comment]
> The PR will not build without main.go and the Makefile doesn't work since the cmd dir structure is not created.

Ooops somehow the `.gitignore` is ignoring `./cmd/...` I'll fix that and amend.

File: Makefile (hunk 1 above)
Comment (prologic) [reply Makefile]: Fixed :) sorry! got missed by `.gitignore`

File: src/main.go (hunk 2 above)
Comment (prologic) [reply src/main.go]: Fixed :)

**Comment by prologic** [comment]
> I see your point of making it internal if it were to be used as a module. But since it will never be used as a module imo, I don't really see the need to change src to internal and create a doc.go file?

Also I see your point too; butt the way I look at it; if you `go doc` this or happen to come across it in  https://godoc.org/ at least you have some documentation about what it is :)

File: Makefile [hunk 3]
```diff
@@ -2,24 +2,24 @@ all: build release clean
...
 	@mkdir release
-	@mkdir gorss
-	@cd src && CC=x86_64-linux-musl-gcc CXX=x86_64-linux-musl-g++ GOARCH=amd64 GOOS=linux CGO_ENABLED=1 go build -ldflags "-linkmode external -extldflags -static -s -w -X main.version=${VERSION}" -o ../release/gorss_linux
-	@cd src && go build -ldflags "-s -w -X main.version=${VERSION}" -o ../release/gorss_osx
-	@cp gorss.conf gorss/
+	@mkdir dist
+	@CC=x86_64-linux-musl-gcc CXX=x86_64-linux-musl-g++ GOARCH=amd64 GOOS=linux CGO_ENABLED=1 go build -ldflags "-linkmode external -extldflags -static -s -w -X $(shell go list)/internal.Version=${VERSION}" -o ../release/gorss_linux ./cmd/gorss/...
+	@go build -ldflags "-s -w -X $(shell go list)/internal.Version=${VERSION}" -o ../release/gorss_osx ./cmd/gorss/...
+	@cp gorss.conf dist/
 	@cp themes/default.theme gorss/
```
Comment (Lallassu) [inline Makefile]: should be `@cp themes/default.theme dist/`

File: Makefile [hunk 4]
```diff
@@ -2,24 +2,24 @@ all: build release clean
...
 	
 
 release:
 	@mkdir release
-	@mkdir gorss
-	@cd src && CC=x86_64-linux-musl-gcc CXX=x86_64-linux-musl-g++ GOARCH=amd64 GOOS=linux CGO_ENABLED=1 go build -ldflags "-linkmode external -extldflags -static -s -w -X main.version=${VERSION}" -o ../release/gorss_linux
-	@cd src && go build -ldflags "-s -w -X main.version=${VERSION}" -o ../release/gorss_osx
-	@cp gorss.conf gorss/
+	@mkdir dist
+	@CC=x86_64-linux-musl-gcc CXX=x86_64-linux-musl-g++ GOARCH=amd64 GOOS=linux CGO_ENABLED=1 go build -ldflags "-linkmode external -extldflags -static -s -w -X $(shell go list)/internal.Version=${VERSION}" -o ../release/gorss_linux ./cmd/gorss/...
```
Comment (Lallassu) [inline Makefile]: Should be ` -o ./release/gorss_linux ./cmd/gorss/...` I guess (./release).

File: Makefile [hunk 5]
```diff
@@ -2,24 +2,24 @@ all: build release clean
...
+	@go build -ldflags "-s -w -X $(shell go list)/internal.Version=${VERSION}" -o ../release/gorss_osx ./cmd/gorss/...
```
Comment (Lallassu) [inline Makefile]: Same here, "./release"...

**Pull request merged**